app/src/androidTest/java/com/example/tictactoe/GameFragmentNavigationTest.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.tictactoe
+
+import androidx.core.os.bundleOf
+import androidx.fragment.app.testing.launchFragmentInContainer
+import androidx.navigation.Navigation
+import androidx.navigation.testing.TestNavHostController
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import com.example.tictactoe.ui.GameFragment
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+@RunWith(JUnit4::class)
```

Comment:
either this isn't needed, or should be `AndroidJUnit4` - you define your test instrumentation runner in your gradle file as `testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"`, so ideally we'd want the class test runner to be the same. i'm a little surprised this test is passing as it is right now, i would think it would fail, but maybe something is using the android junit 4 runner instead under the hood

Reply:
Interestingly when I tried switching runners I got this message:

<img width="803" alt="runner_error_msg_02:10:22" src="https://user-images.githubusercontent.com/62267982/153494765-61a3fcec-229a-4e78-b67d-dd582faa2329.png">

I'm not certain why.

I'll fiddle with changing the gradle file after I figure out my navigation tests.